.github/workflows/nightly-prerelease.yaml

```diff
@@ -0,0 +1,179 @@
+name: Nightly Pre-release
+on:
+  # TODO: Enable schedule after testing
+  # schedule:
+  #   # Run at 7 AM UTC daily
+  #   - cron: "0 07 * * *"
+  workflow_dispatch:
+    inputs:
+      dry-run:
+        description: "Dry run"
+        required: false
+        default: true
+        type: boolean
```

Comment:
This is currently a dry-run only workflow because we will have to setup a GitHub app to actually make the tag rather than using a `GITHUB_TOKEN`

> If you do want to trigger a workflow from within a workflow run, you can use a GitHub App installation access token or a personal access token instead of GITHUB_TOKEN to trigger events that require a token.